nc/source/math/Quaternion.cpp

```diff
@@ -0,0 +1,47 @@
+#include "Quaternion.h"
+#include "directx/math/DirectXMath.h"
+
+namespace nc
+{
+    Quaternion::Quaternion(const DirectX::XMFLOAT4& xmf4) noexcept
```

Comment:
If you put validation in this class instead (throw if this would create an invalid Quaternion) then you do not have to have the responsibility of testing for Quaternion validitiy off shifted to all of the consumers of Quaternion.

Reply:
We do something like that at work: in CTORs - throw if args are invalid